windows/src/init.c

```diff
@@ -136,14 +136,7 @@ libvchan_t *libvchan_client_init(int domain, int port) {
 
     status = WaitForSingleObject(path_watch_event, 100);
     if (status != WAIT_OBJECT_0) {
-        Log(XLL_WARNING, "Wait for xenstore (1) failed: 0x%x", GetLastError());
-        // don't fail completely yet, if we can read the store values we're ok
-    }
-
-    // wait two times because Xen always signals the watch immediately after creation
```

Comment:
Are you sure this second wait is not needed? If the xenstore path already exists at this point, neither of them is needed, but if doesn't only that second wait actually waits for an xenstore event, no?

Reply:
> Are you sure this second wait is not needed? If the xenstore path already exists at this point, neither of them is needed, but if doesn't only that second wait actually waits for an xenstore event, no?

I think the comment about Xen signaling the event immediately was windows pvdriver specific and this behavior was changed to eliminate this first (useless) event signal. I'll double check, but the Linux code doesn't perform two waits. Also it probably would be nice to have some unit tests for such things...

Reply:
Actually, just waiting twice is wrong. There should be a check if the path exists between them, and the second wait should be only in case it doesn't.

You can test things by doing qrexec and pausing the other VM while qrexec prompt is visible. Not sure about the direction of the call for this specific case, but you'll figure it out - just two possibilities.